Modules/_testcapimodule.c

```diff
@@ -3231,6 +3231,86 @@ test_weakref_capi(PyObject *Py_UNUSED(module), PyObject *Py_UNUSED(args))
 }
 
 
+static PyObject *
+sys_getattr(PyObject *Py_UNUSED(module), PyObject *name)
+{
+    NULLABLE(name);
+    PyObject *result = PySys_GetAttr(name);
+    if (result == NULL && PyErr_Occurred()) {
+        return NULL;
+    }
+    if (result == NULL) {
+        result = PyExc_AttributeError;
+        Py_INCREF(PyExc_AttributeError);
```

Comment:
I don't understand this code path. The function must return NULL *and* raise an exception if the attribute does not exist. This code path must never be reached according to the API doc.